### PR TITLE
Adding tests

### PR DESCRIPTION
**Pull request by fubhy**

### CODE REVIEW

File: src/contracts/Accounting.test.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  beforeAll(() => {
+    // TODO: This should be replaced with a local ganache test environment using proper test fixtures.
+    const client = new Eth(new HttpProvider('https://mainnet.melonport.com'));
+    environment = new Environment(client, deployment);
+    accounting = new Accounting(environment, '0x1b66598123fefb8759340d4ea6e4b070c4fc4315');
+  });
+
+  it('should return the default share price', async () => {
+    const result = await accounting.defaultSharePrice();
+    expect(result).toStrictEqual(new BigNumber('1000000000000000000'));
```
Comment (fubhy) [inline src/contracts/Accounting.test.ts]: Should use bignumber.js equality checks here.

```typescript
expect(a.isEqualTo(b)).toBe(true);
```

File: src/contracts/Accounting.test.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  });
+
+  it('should return WETH address as native asset', async () => {
+    const result = await accounting.nativeAsset();
+    expect(toChecksumAddress(result)).toBe(
+      toChecksumAddress(environment.deployment.thirdPartyContracts.tokens[4].address),
```
Comment (fubhy) [inline src/contracts/Accounting.test.ts]: Use ```Token.findToken(environment, 'TOKEN');``` for this.

File: src/contracts/Hub.test.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  });
+
+  it('should return the name as "miFund"', async () => {
+    const result = await hub.name();
+    expect(result).toBe('miFund');
+  });
+
+  it('should return the managers as "0x1e1c1ba503cc84b1bcf0151c68a8b5ddc90e4a2e"', async () => {
+    const result = await hub.manager();
+    expect(toChecksumAddress(result)).toBe(toChecksumAddress('0x1e1c1ba503cc84b1bcf0151c68a8b5ddc90e4a2e'));
```
Comment (fubhy) [inline src/contracts/Hub.test.ts]: I think this should use toEqual. Not sure though. Can you check the Jest documentation and/or stackoverflow?

File: src/contracts/Hub.test.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  beforeAll(() => {
+    // TODO: This should be replaced with a local ganache test environment using proper test fixtures.
+    const client = new Eth(new HttpProvider('https://mainnet.melonport.com'));
+    environment = new Environment(client, deployment);
+    hub = new Hub(environment, '0x7362d4dd194cf5174748317da1e960723b78f28c');
+  });
+
+  it('should return the name as "miFund"', async () => {
+    const result = await hub.name();
+    expect(result).toBe('miFund');
```
Comment (fubhy) [inline src/contracts/Hub.test.ts]: Same here. Not sure if toBe() or toEqual() are correct for this.

File: src/contracts/Hub.test.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+  it('should return the creation time', async () => {
+    const result = await hub.creationTime();
+    expect(result).toBeInstanceOf(Date);
+    expect(result.getTime()).toBeLessThan(Date.now());
+  });
+
+  it('should return the routes', async () => {
+    routes = await hub.routes();
+    expect(routes).toHaveProperty('accounting');
```
Comment (fubhy) [inline src/contracts/Hub.test.ts]: I think Jest has a method to assert an object shape. Can you check?

File: src/contracts/Hub.test.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+import { HttpProvider } from 'web3-providers';
+import { Environment } from '../Environment';
+import deployment from '../deployments/mainnet';
+import { Hub, FundRoutes } from './Hub';
+import { toChecksumAddress } from 'web3-utils';
+
+describe('Hub', () => {
+  let environment: Environment;
+  let hub: Hub;
+  let routes: FundRoutes;
```
Comment (fubhy) [inline src/contracts/Hub.test.ts]: You don't need to create a shared `routes` variable up here unless you want to share it across different tests.

File: src/contracts/Hub.test.ts (hunk 4 above)
Comment (iherger) [reply src/contracts/Hub.test.ts]: For primitive types, it's doesn't make a difference, apparently, see https://stackoverflow.com/a/27929503

File: src/contracts/Hub.test.ts (hunk 3 above)
Comment (iherger) [reply src/contracts/Hub.test.ts]: Probably better to do:
` expect(sameAddress(result, '0x1e1c1ba503cc84b1bcf0151c68a8b5ddc90e4a2e')).toBe(true);`

File: src/contracts/Accounting.test.ts (hunk 1 above)
Comment (iherger) [reply src/contracts/Accounting.test.ts]: Yes.

File: src/contracts/Accounting.test.ts (hunk 2 above)
Comment (iherger) [reply src/contracts/Accounting.test.ts]: `Token.findToken(environment, 'TOKEN');` doesn't exist, as far as I am aware. Shall I write it?

Comment (fubhy) [reply src/contracts/Accounting.test.ts]: Sorry, it's Token.findDefinition. Inherited from AbstractToken.

File: src/contracts/Hub.test.ts (hunk 4 above)
Comment (fubhy) [reply src/contracts/Hub.test.ts]: Okay perfect then.

File: src/contracts/Hub.test.ts (hunk 5 above)
Comment (iherger) [reply src/contracts/Hub.test.ts]: Now using:
```ts
    expect(routes).toMatchObject<FundRoutes>({
      accounting: expect.any(String),
      engine: expect.any(String),
      feeManager: expect.any(String),
      mlnToken: expect.any(String),
      participation: expect.any(String),
      policyManager: expect.any(String),
      priceSource: expect.any(String),
      registry: expect.any(String),
      shares: expect.any(String),
      trading: expect.any(String),
      vault: expect.any(String),
      version: expect.any(String),
    });
```

File: src/contracts/Hub.test.ts (hunk 6 above)
Comment (iherger) [reply src/contracts/Hub.test.ts]: Actually sharing it between two tests.

File: src/contracts/Hub.test.ts (hunk 5 above)
Comment (fubhy) [reply src/contracts/Hub.test.ts]: Perfect

File: src/contracts/Hub.test.ts (hunk 6 above)
Comment (fubhy) [reply src/contracts/Hub.test.ts]: Alright then

**Pull request merged**